contrib/implements/internal/implements/gosrc.go

```diff
@@ -106,7 +106,7 @@ func readDocComment(fdec *ast.FuncDecl, gfunc *goFunction) {
 }
 
 func isPublic(gfunc *goFunction) bool {
-	return ast.IsExported(gfunc.shortName)
+	return ast.IsExported(gfunc.shortName) && ast.IsExported(gfunc.fullName)
```

Comment:
Isn't that redundant? If `fullName` is exported, then `shortName` is also exported, right?

Reply:
No, fullName contains the name of the type when the function is a receiver method. So shortName might be "fooFunc" but full name will be "coolType.fooFunc". And IsExported is very simple and basically just checks if the first character of the string is uppercase or not, so calling `IsExported(fullName)` does end up checking the name of the type (only IIUC).